sdk/Trace/ExporterFactory.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\HttpFactory;
+
+class ExporterFactory
+{
+    private $name;
+    private $_allowedExporters = ['Jaeger' => true, 'Zipkin' => true, 'Newrelic' => true, 'Otlp' => true, 'Otlpgrpc' => true, 'Zipkintonewrelic' => true];
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+      * Selects the correct Exporter via the configuration string
+      * Currently only supports Jaeger and Zipkin exporters
+      *
+      * @param string $configurationString String containing unextracted information for Exporter creation
+      */
+    public function fromConnectionString(string $configurationString)
+    {
+        $strArr = htmlentities($configurationString);
```

Comment:
Ok so I made a mistake here, we should never escape input, its bad practice because it permanently ties that string to its final output format. So from what I understand you are usually only expecting some type of identifier with a url at the end. So what you want to do is filter the input to only be in the format you expect. Look at these filters [here](https://www.w3schools.com/php/filter_validate_url.asp) and [here](https://www.php.net/manual/en/filter.filters.sanitize.php). You can also make your own filter through a regular expression which might be what you want. This will make sure that you are getting input only in the format that you desire.

From what I see, I don't believe these input values ever are on some type of html page so you can exclude the use of html entities. If it actually does end up on an html page you would use htmlentities() when your outputting it, not on input. So I think you can safely exclude htmlentities() since its just a url but definitely add the format filter you want instead